lib/NavLink.js

```diff
@@ -6,20 +6,68 @@
 'use strict';
 
 var React = require('react');
-var NavLink;
-var navigateAction = require('../actions/navigate');
+var navigateAction = require('./navigateAction');
 var debug = require('debug')('NavLink');
 var objectAssign = require('object-assign');
+var RouterMixin = require('./RouterMixin');
 
-NavLink = React.createClass({
+var NavLink = React.createClass({
     displayName: 'NavLink',
-    propTypes: {
-        context: React.PropTypes.object.isRequired
+    contextTypes: {
+        executeAction: React.PropTypes.func,
+        getStore: React.PropTypes.func
+    },
+    getInitialState: function () {
+        var href = this._getHrefFromProps(this.props);
+        return {
+            href: this._getHrefFromProps(this.props),
+            isActive: this._getRouteStore().isActive(href)
+        };
+    },
+    componentWillReceiveProps: function (nextProps) {
+        var href = this._getHrefFromProps(nextProps);
+        this.setState({
+            href: this._getHrefFromProps(nextProps),
+            isActive: this._getRouteStore().isActive(href)
+        });
+    },
+    componentDidMount: function () {
+        this._getRouteStore().addChangeListener(this._onStoreChange);
```

Comment:
One thing to call out, there could be more than 10 NavLink's on the page, which will trigger the max listener count warning with event emitter.